src/network/network_client.cpp

```diff
@@ -1344,27 +1344,22 @@ bool NetworkValidateClientName()
 }
 
 /**
- * Send the server our name.
+ * Send the server our name as callback from the setting.
+ * @param newname The new client name.
  */
-void NetworkUpdateClientName()
+void NetworkUpdateClientName(const std::string &client_name)
 {
 	NetworkClientInfo *ci = NetworkClientInfo::GetByClientID(_network_own_client_id);
-
 	if (ci == nullptr) return;
-	/* There is no validation on string settings, it is actually a post change callback.
-	 * This method is called from that post change callback. So, when the client name is
-	 * changed via the console there is no easy way to prevent an invalid name. Though,
-	 * we can prevent it getting sent here. */
-	if (!NetworkValidateClientName()) return;
 
 	/* Don't change the name if it is the same as the old name */
-	if (_settings_client.network.client_name.compare(ci->client_name) != 0) {
+	if (client_name.compare(ci->client_name) != 0) {
```

Comment:
Similar as earlier: `client_name != ci->client_name`? Or does that do something else?

I see more of these cases throughout this PR, so will await the answer before I mark the other places :D

Reply:
It would function the same, but this change is (now) easier for me.